routes/runtime/github.rb

```diff
@@ -26,11 +26,12 @@ class Clover
       dataset = repository.cache_entries_dataset
         .exclude(committed_at: nil)
         .where(version: version, scope: scopes)
+        .order(Sequel.case(scopes.map.with_index { |scope, idx| [{scope:}, idx] }.to_h, scopes.length))
 
       entry = dataset
         .where(key: keys)
-        .all
-        .min_by { keys.index(_1.key) + (scopes.index(_1.scope) * keys.size) }
+        .order_prepend(Sequel.case(keys.map.with_index { |key, idx| [{key:}, idx] }.to_h, keys.length))
```

Comment:
Should we use `order_append` instead of `order_prepend`?

Let's think a scenario like this:
```
keys = [a, b]
scopes = [dev, main]

cache_entries
1. {key: a, scope: main}
2. {key: b, scope: main}
3. {key: b, scope: dev}
```

The old code returns (3). If I'm not mistaken, the query returns (1) because it uses `ORDER BY key, scope`.

Reply:
Yep, you are correct.  I got confused because of the order of the `min_by`.  I'll fix this.

Reply:
I pushed a commit for this, with a new comment describing the prioritization.